src/flowcean/transforms/one_hot.py

```diff
@@ -0,0 +1,71 @@
+import logging
+from collections.abc import Iterable
+from typing import override
+
+import polars as pl
+
+from flowcean.core import Transform
+
+logger = logging.getLogger(__name__)
+
+
+class OneHot(Transform):
+    """Transforms integer features into a set of binary one-hot features.
+
+    Transforms integer features into a set of binary one-hot features. The
+    original integer features are dropped and are not part of the resulting
+    data frame.
+
+    As an example consider the following data
+
+    feature |
+    :------:|
+     0      |
+     1      |
+     2      |
+     1      |
+     5      |
+
+    When the one-hot transformation is applied, the result is as follows
+
+    feature_0 | feature_1 | feature_2 | feature_5
+    ----------|-----------|-----------|----------
+     1        | 0         | 0         | 0
+     0        | 1         | 0         | 0
+     0        | 0         | 1         | 0
+     0        | 1         | 0         | 0
+     0        | 0         | 0         | 1
+    """
+
+    def __init__(self, features: Iterable[str]) -> None:
+        """Initializes the One-Hot transform.
+
+        Args:
+            features: The features to transform.
+        """
+        self.features = features
+
+    @override
+    def transform(self, data: pl.DataFrame) -> pl.DataFrame:
+        for feature in self.features:
+            # Check if the feature is an integer feature
+            if not data.schema[feature].is_integer():
```

Comment:
Why only integer? Isn't this applicable to all categorical values?

Reply:
It is. But we have no way in the framework to handle categorical values besides ints. We could - in theory - use `unique()` to get a set of all present values and then transform based on those... But I'm not sure I like the idea.

What I just notice: We should convert this transform into a `Supervised Learner` like `Standardize`. Otherwise results may not be consistent when calling the transform for different data frames.

Reply:
I am not sure whether this works... This would add additional columns/features to the dataframe while iterating the environment?! Does that work? The learners would have to cope with changing feature names...

For example, if the first row in an incremental environment only shows a single value, then the one hot encoding is useless, and there still would be a single feature, but as soon as the second categorical variant arrives, the one-hot encoding would add an additional feature...

Reply:
But exactly the described situation does not work for an incremental environment. If this is fitted incrementally, the input features change while training....

Reply:
I think this might be the only, or at least the best option I can currently think of... Let the user specify the categorical variants of each feature one wants to encode

Reply:
It's the same problem as with `Standardize`. If we adjust to little data we might get unwanted results.
But for `OneHot` I would add the categories as an optional argument. If you have enough data / an offline environment, automatically determining the categories feels like a worthwhile addition.